kafka/consumer_test.go

```diff
@@ -429,3 +431,277 @@ func TestConsumerLog(t *testing.T) {
+	if err != nil {
+		t.Fatalf("Failed to subscribe to topic %s: %s\n", topic, err)
+		return
+	}
+	wg.Add(1)
+	go testPoll(c2, closeChan, t, wg)
+	testConsumerWaitAssignment(c2, t)
+
+	wrapRebalancecb3 := wrapRebalanceCb(&assignedEvents3, &revokedEvents3, t)
+	err = c3.Subscribe(topic, wrapRebalancecb3)
+	if err != nil {
+		t.Fatalf("Failed to subscribe to topic %s: %s\n", topic, err)
+		return
+	}
+	wg.Add(1)
+	go testPoll(c3, doneChan, t, wg)
+	testConsumerWaitAssignment(c3, t)
+
+	closeChan <- true
+	close(closeChan)
+	c2.Close()
+
+	c2, err = NewConsumer(&conf2)
+	if err != nil {
+		t.Fatalf("%s", err)
+	}
+
+	wrapRebalancecb4 := wrapRebalanceCb(&assignedEvents4, &revokedEvents4, t)
+	err = c2.Subscribe(topic, wrapRebalancecb4)
+	if err != nil {
+		t.Fatalf("Failed to subscribe to topic %s: %s\n", topic, err)
+		return
+	}
+	wg.Add(1)
+	go testPoll(c2, doneChan, t, wg)
+	testConsumerWaitAssignment(c2, t)
+
+	doneChan <- true
+	close(doneChan)
+
+	c3.Close()
+	c2.Close()
+	c1.Close()
+
+	wg.Wait()
+
+	// Wait 30s to make sure no revokedEvents happens
+	time.Sleep(30 * time.Second)
+
+	if atomic.LoadInt32(&assignedEvents1) != 3 {
+		t.Fatalf("3 assignedEvents are Expected to happen for the first consumer, but %d happened\n",
+			atomic.LoadInt32(&assignedEvents1))
+	}
+
+	if atomic.LoadInt32(&revokedEvents1) != 2 {
+		t.Fatalf("2 revokedEvents are Expected to happen for the first consumer, but %d happened\n",
+			atomic.LoadInt32(&revokedEvents1))
+	}
+
+	if atomic.LoadInt32(&assignedEvents2) != 2 {
+		t.Fatalf("2 assignedEvents are Expected to happen for the second consumer, but %d happened\n",
+			atomic.LoadInt32(&assignedEvents2))
+	}
+	if atomic.LoadInt32(&revokedEvents2) != 1 {
+		t.Fatalf("1 revokedEvents is Expected to happen for the second consumer, but %d happened\n",
+			atomic.LoadInt32(&revokedEvents2))
+	}
+
+	if atomic.LoadInt32(&assignedEvents3) != 1 {
+		t.Fatalf("1 assignedEvents is Expected to happen for the third consumer, but %d happened\n",
+			atomic.LoadInt32(&assignedEvents3))
+	}
+	if atomic.LoadInt32(&revokedEvents3) != 0 {
+		t.Fatalf("0 revokedEvents is Expected to happen for the third consumer, but %d happened\n",
+			atomic.LoadInt32(&revokedEvents3))
+	}
+
+	if atomic.LoadInt32(&assignedEvents4) != 1 {
+		t.Fatalf("1 assignedEvents is Expected to happen for the rejoined consumer(originally second consumer), but %d happened\n",
+			atomic.LoadInt32(&assignedEvents4))
+	}
+	if atomic.LoadInt32(&revokedEvents4) != 0 {
+		t.Fatalf("0 revokedEvents is Expected to happen for the rejoined consumer(originally second consumer), but %d happened\n",
+			atomic.LoadInt32(&revokedEvents4))
+	}
+}
+
+func testConsumerWaitAssignment(c *Consumer, t *testing.T) {
+	run := true
+	for run {
+		assignment, err := c.Assignment()
+		if err != nil {
+			t.Fatalf("Assignment failed: %s", err)
+		}
+
+		if len(assignment) != 0 {
+			fmt.Printf("%v Assigned partitions are: %v\n", c, assignment)
```

Comment:
Did you consider using `t.Logf` in tests instead of `fmt.Printf `? This makes debugging logs better for debugging and easy to disable, by not using the `-v` flag.